### ISSUE TITLE
Should not lint on dynamically added R6 method

### ISSUE THREAD

**Issue by radbasa** [Appsilon/box.linters#24]
## box.linters version

0.0.0.9006

## Source code to lint

```R
box::use(
  R6[R6Class],
)

SomeClass <- R6Class("SomeClass",   # nolint
  public = list()
)

SomeClass$set("public", "x", function() {
  "X"
})

s <- SomeClass$new()
s$x()
```

## Lint command

```R
lintr::lint(code, linters = lintr::linters_with_defaults(defaults = box.linters::rhino_default_linters))
```

## Lint result

```
<text>:15:1: warning: [box_usage_linter] <package/module>$function does not exist.
s$x()
^~~
```

## Previous issues

- #22 
- #23 

## Expected result

Should not lint on dynamically added function.
